modules/store/spec/selector.spec.ts

```diff
@@ -126,13 +114,27 @@ describe('Selectors', () => {
       expect(projectFn).toHaveBeenCalledTimes(2);
     });
 
+    it('should not memoize last successful projection result in case of error', () => {
+      const firstState = { ok: 'a' };
+      const secondState = { faulty: 'b' };
+      const fail = () => {
+        throw new Error();
+      };
+      const selector = createSelector(
+        identity,
+        (s: any) => (s.ok ? s.ok : fail())
+      );
+
+      selector(firstState);
+
+      expect(() => selector(secondState)).toThrow(new Error());
+      expect(() => selector(secondState)).toThrow(new Error()); // fails here, returns 'a'
```

Comment:
Comment can be left out, as you've fixed it

```suggestion
      expect(() => selector(secondState)).toThrow(new Error());
```

Reply:
Can we also expect the selector to be called 3 times here? To make sure it does re-execute the projector function.

To be completely sure we should also re-execute the selector with `firstState`, to make sure it still has the previous result memoized.

Reply:
Not sure if I understood you correctly and if it is any better now, please check the updated commit. To verify how many times the selector and its projector are called, I had to introduce some spying.

It now verifies that the selector is called 4 times (this should be obvious as those calls happen directly in the test case - probably not required). The projector is called 3 times: 1x for `firstState`, and 2x for `secondState`. Because the originally memoized value for `firstState` is still valid after 2 failures, the projector is not evaluated again in the end when selecting from that state.

Reply:
This is exactly what I meant, thanks 👍 
The test to verify the selector has been called 4 times, may seem redundant but this makes it clear imho.
Thanks!

Could you in the future not force push your new commits and just create a new commit? This makes it easier for us to verify the changes to the original commit. We'll squash the commits into one when we merge to master.

Reply:
Sorry, I will make sure to preserve a proper commit history for future PRs.

By the way I was surprised that some (untouched, unrelated) code got reformatted. But I believe this was done by a pre-commit hook and not my IDE.